module/friendship/app/query/list_common_friends.go

```diff
@@ -0,0 +1,99 @@
+package query
+
+import (
+	"context"
+
+	"github.com/phantranhieunhan/s3-assignment/common"
+	"github.com/phantranhieunhan/s3-assignment/common/logger"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
+)
+
+type ListCommonFriendsRepo interface {
+	GetFriendshipByUserIDAndStatus(ctx context.Context, userID string, status ...domain.FriendshipStatus) (domain.Friendships, error)
+}
+
+type ListCommonFriendsHandler struct {
+	repo     ListCommonFriendsRepo
+	userRepo UserRepo
+}
+
+func NewListCommonFriendsHandler(repo ListCommonFriendsRepo, userRepo UserRepo) ListCommonFriendsHandler {
+	return ListCommonFriendsHandler{
+		repo:     repo,
+		userRepo: userRepo,
+	}
+}
```

Comment:
repository and handler should be introduced in separated files in dedicated corresponding folder